src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -197,6 +197,9 @@ public KNNVectorFieldMapper build(BuilderContext context) {
             final CopyTo copyToBuilder = copyTo.build();
             final Explicit<Boolean> ignoreMalformed = ignoreMalformed(context);
             final Map<String, String> metaValue = meta.getValue();
+
+            updateStats(knnMethodContext);
```

Comment:
I think this will miss the case if the field uses a model.

Reply:
I'm not sure why it's so? In the same method we do have a case [when model is in use](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java#L234), and this happens after that we've updated the stat. Do you mean we have completely different code path that doesn't even go [build()](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java#L185) method

Reply:
If we are using a model for the field, it wont indicate that the correct engine_field_in_use because knnmethodcontext will be null: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java#L233

Reply:
I see what you mean. In this case can we assume the engine implicitly from the fact that model id is set? Seems that faiis is the only engine in this case. Alternatively we can skip the stat setting when knnmethodcontext and engine are not set.
I also wonder why we do have "nmslib" as default engine then? When working with the code I was thinking we need to use DEFAULT in case it's not set explicitly in request.

Reply:
We could actually get the engine from the model metadata. We get the metadata [here](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/ModelFieldMapper.java#L49) but I think we could get it earlier if we need to.

Reply:
> I also wonder why we do have "nmslib" as default engine then?

Right, when engine is not specified it is assumed to be nmslib. However, with models, the engine is specified with the model metadata.

Reply:
ack, will change in next commit